Sources/StreamChatUI/Components.swift

```diff
@@ -302,6 +302,18 @@ public struct Components {
 
     /// The view that shows a number of unread messages in channel.
     public var channelUnreadCountView: ChatChannelUnreadCountView.Type = ChatChannelUnreadCountView.self
+    
+    /// The view that is displayed when there are no channels on the list, i.e. when is on empty state.
+    public var channelListEmptyView: ChatChannelListEmptyView.Type = ChatChannelListEmptyView.self
+    
+    /// View that shows that some error occurred on ChatChannelList.
+    public var channelListErrorView: ChatChannelListErrorView.Type = ChatChannelListErrorView.self
```

Comment:
As described above, these ones should be `protocol: UIView` 👍

Reply:
@nuno-vieira as far as I can see, all the views are concrete open classes, not protocols. I think it would be better to keep it consistent, and if we do your suggested change, to do it for all views (in a separate ticket).

Reply:
These views are special ones because the customer most likely can reuse their existing state views, so it is important that they are not concrete classes. As far as I remember from the previous PRs, this was a requirement and not a nice to have.

Reply:
Yes, that makes sense. Cool, let's roll with that 👍

Reply:
As discussed earlier today, we are going to move forward with the concrete class/inheritance approach since is consistent with the rest of our components but I added documentation on how to customize either of the classes so implementers can easily customize them 👍🏻